src/DotNetBoilerplate.Api/Reservations/CreateReservationEndpoint.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.ComponentModel.DataAnnotations;
+using DotNetBoilerplate.Application.Exceptions;
+using DotNetBoilerplate.Application.Reservations.CreateReservation;
+using DotNetBoilerplate.Shared.Abstractions.Commands;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DotNetBoilerplate.Api.Reservations;
+
+public class CreateReservationEndpoint : IEndpoint
+{
+    public static void Map(IEndpointRouteBuilder app)
+    {
+        app.MapPost("/event/{eventId:guid}/reservations/", Handle)
+            .RequireAuthorization()
+            .WithSummary("Creates new reservation");
+    }
+
+    private static async Task<Results<Ok<Response>, NotFound>> Handle(
+        [FromRoute] Guid eventId,
+        [FromServices] ICommandDispatcher commandDispatcher,
+        CancellationToken ct
+    )
+    {
+        var command = new CreateReservationCommand(
+            eventId
+            );
+
+
+
+        try
```

Comment:
Tutaj nie musimy używać try-catch - po to mamy zaimplementowany custom middleware żeby naszego exceptiona jakos opakował w ładny komunikat